runners/flink/1.8/src/main/java/org/apache/beam/runners/flink/translation/types/CoderTypeSerializer.java

```diff
@@ -49,20 +50,18 @@
    * org.apache.beam.sdk.io.FileSystems} registration needed for {@link
    * org.apache.beam.sdk.transforms.Reshuffle} translation.
    */
-  @SuppressWarnings("unused")
-  private final @Nullable SerializablePipelineOptions pipelineOptions;
+  private final SerializablePipelineOptions pipelineOptions;
 
-  public CoderTypeSerializer(Coder<T> coder) {
-    Preconditions.checkNotNull(coder);
-    this.coder = coder;
-    this.pipelineOptions = null;
-  }
+  private final boolean fasterCopy;
```

Comment:
Small suggestion, can we name this `zeroCopy`, as that is actually what it is.

Reply:
I thought about `zeroCopy` as a name, but for me this is associated with the [lower level concept](https://en.wikipedia.org/wiki/Zero-copy) and I thought it might create confusion. The name `fasterCopy` came from the idea that it is making the copying faster, but I am very much open to other names!

Reply:
Okay, what about `disableValueClone`, I think that `fasterCopy` is a little misleading as well. Faster than what? And why it was slower before? :) It would be good for the flag to describe direct effect (disable, enable something), not a consequence (being faster).

Reply:
Very much agree with the name being descriptive. How about we go more opinionated, something like `disableExcessCopy`? There where some back and forth on the mailing list about this point, so I don't want to sneak in my opinion if that is not right. What do you think @mxm ?

Reply:
Folks, this is an internal boolean flag which is only used at a single place. If in doubt, one can clearly see what it does. We could add a `/** */` Javadoc field comment for it if you will. IMHO the name is really personal preference and I have nothing against `fasterCopy` as it clearly indicates that the `copy()` method will return faster. Nothing semantically wrong about that. `zeroCopy` is not 100% true because we will perform a copy for primitive types but please let's not argue about that. `DisableExcessCopy` is too opinionated IMHO 😇

Reply:
I agree with `DisableExcessCopy` being too opinionated. My vote is for `fasterCopy`, but it is not a hill I will die on :P

Reply:
To make it clear - although I put my comment to the internal field, I was having in mind mostly the flag in PipelineOptions. These should be aligned.

Reply:
Agree that this is not 100% important, but if possible, we should make options passed in PipelineOptions the most self-explanatory as possible, because these options are user-facing.

Reply:
Ah, yeah! I'll change it right away, I missed that!

Reply:
I've been talking about the user facing flag from the beginning, so I understood you.